individual-files/Cart.java

```diff
@@ -0,0 +1,52 @@
+package sessions;
+
+import javax.servlet.http.*;
+import java.util.Vector;
+import java.util.Enumeration;
+
+public class Cart {
+    Vector v = new Vector();
+    String submit = null;
+    String item = null;
+   
+    private void addItem(String name) {
+        v.addElement(name);
+    }
+
+    private void removeItem(String name) {
+        v.removeElement(name);
+    }
+
+    public void setItem(String name) {
+        item = name;
+    }
+   
+    public void setSubmit(String s) {
+        submit = s;
+    }
+
+    public String[] getItems() {
+        String[] s = new String[ v.size()];
+        v.copyInto(s);
+        return s;
+    }
+   
+    public void processRequest(HttpServletRequest request) {
+        if (submit == null)
```

Comment:
Unintended Behavior (U) - Within processRequest() function, if submit is null then an item is added to the vector. This is unintended behavior for the program, though without the appropriate comments this cannot be confirmed. 

Rank 2 - Fix as part of planned debugging sessions